presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorColumnIdentity.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor;
+
+import com.facebook.presto.spi.ColumnIdentity;
+import com.facebook.presto.spi.PrestoException;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.io.BaseEncoding.base16;
+import static com.google.common.primitives.Longs.fromByteArray;
+import static com.google.common.primitives.Longs.toByteArray;
+import static java.lang.String.format;
+import static java.lang.System.arraycopy;
+import static java.util.Arrays.copyOfRange;
+import static java.util.Objects.hash;
+import static java.util.Objects.requireNonNull;
+
+public class RaptorColumnIdentity
+        implements ColumnIdentity
+{
+    private static final long CURRENT_VERSION = 1;
+
+    private final long version;
+    private final long columnId;
+
+    public RaptorColumnIdentity(long columnId)
+    {
+        this(CURRENT_VERSION, columnId);
+    }
+
+    private RaptorColumnIdentity(long version, long columnId)
+    {
+        checkArgument(version > 0, "version must be a positive integer");
+        this.version = version;
+        this.columnId = columnId;
+    }
+
+    public static RaptorColumnIdentity of(byte[] bytes)
+    {
+        requireNonNull(bytes, "bytes for ColumnIdentity is null");
+
+        if (bytes.length > Long.BYTES) {
```

Comment:
I'd use this simpler form.  I'd also use a little endian encoding for the column Id
```java
if (bytes.lenght == 5)
    throw new PrestoException(GENERIC_INTERNAL_ERROR, format("RaptorColumnIdentity is corrupt: %s", base16().upperCase().encode(bytes)));
}
byte version = bytes[0];
long columnId = Longs.fromBytes(bytes[8], bytes[7], bytes[6], bytes[5], bytes[4], bytes[3], bytes[2], bytes[1]);
return new RaptorColumnIdentity(version, columnId);